scribus/text/storytext.cpp

```diff
@@ -1516,20 +1519,14 @@ int StoryText::firstWord()
 
 int StoryText::nextWord(int pos)
 {
-	int len = length();
-	//move to text
-	pos = qMin(len, pos+1);
-	//while not at the end, and while we don't find a word boundary, move to the next character
-	while (pos < len  && wordBoundaries.indexOf(text(pos)) < 0)
-		++pos;
-	//while not at the end, and while we find a word boundary, move to the next character
-	while (pos < len  && wordBoundaries.indexOf(text(pos)) >= 0)
-		++pos;
-	//if we didn't get to the end, return current position, otherwise return the end.
-	if (pos < len)
+	BreakIterator* it = getWordIterator();
+	if (!it)
 		return pos;
-	else
-		return len;
+
+	it->setText(plainText().utf16());
+	int len = length();
+	int nxt = it->following(pos);
+	return nxt;
 }
 
 int StoryText::prevWord(int pos)
```

Comment:
Either port it for completeness sake, since it might get be used in the future and it will be wrong then, or completely remove it. I prefer keeping it for API parity, it does not hurt much.